compiler/frontend-spark/src/main/java/edu/snu/onyx/compiler/frontend/spark/JavaRDD.java

```diff
@@ -0,0 +1,196 @@
+   * @return the Java RDD with the initialized source vertex.
+   */
+  JavaRDD<T> setSource(final Iterable<T> initialData) {
+    final DAGBuilder<IRVertex, IREdge> builder = new DAGBuilder<>(dag);
+
+    final IRVertex initializedSourceVertex = new InitializedSourceVertex<>(initialData);
+    initializedSourceVertex.setProperty(ParallelismProperty.of(parallelism));
+    builder.addVertex(initializedSourceVertex, loopVertexStack);
+
+    return new JavaRDD<>(this.sparkContext, this.parallelism,
+        builder.buildWithoutSourceSinkCheck(), initializedSourceVertex);
+  }
+
+  /**
+   * Map transform.
+   * @param func function to apply.
+   * @param <O> output type.
+   * @return the JavaRDD with the DAG.
+   */
+  public <O extends Serializable> JavaRDD<O> map(final SerializableFunction<T, O> func) {
+    final DAGBuilder<IRVertex, IREdge> builder = new DAGBuilder<>(dag);
+
+    final IRVertex mapVertex = new OperatorVertex(new MapTransform<>(func));
+    mapVertex.setProperty(ParallelismProperty.of(parallelism));
+    builder.addVertex(mapVertex, loopVertexStack);
+
+    final IREdge newEdge = new IREdge(getEdgeCommunicationPattern(lastVertex, mapVertex),
+        lastVertex, mapVertex, new SparkCoder(kryoSerializer));
+    newEdge.setProperty(KeyExtractorProperty.of(new SparkKeyExtractor()));
+    builder.connectVertices(newEdge);
+
+    return new JavaRDD<>(this.sparkContext, this.parallelism, builder.buildWithoutSourceSinkCheck(), mapVertex);
+  }
+
+  /////////////// ACTIONS ///////////////
+
+  private static final AtomicInteger RESULT_ID = new AtomicInteger(0);
+
+  /**
+   * This method is to be removed after a result handler is implemented.
+   * @return a unique integer.
+   */
+  public static Integer getResultId() {
+    return RESULT_ID.getAndIncrement();
+  }
+
+  /**
+   * Reduce action.
+   * @param func function (binary operator) to apply.
+   * @return the result of the reduce action.
+   */
+  public T reduce(final SerializableBinaryOperator<T> func) {
+    // save result in a temporary file
+    final String resultFile = System.getProperty("user.dir") + "/reduceresult";
+    final DAGBuilder<IRVertex, IREdge> builder = new DAGBuilder<>(dag);
+
+    final IRVertex reduceVertex = new OperatorVertex(new ReduceTransform<>(func, resultFile));
+    reduceVertex.setProperty(ParallelismProperty.of(parallelism));
+    builder.addVertex(reduceVertex, loopVertexStack);
+
+    final IREdge newEdge = new IREdge(getEdgeCommunicationPattern(lastVertex, reduceVertex),
+        lastVertex, reduceVertex, new SparkCoder(kryoSerializer));
+    newEdge.setProperty(KeyExtractorProperty.of(new SparkKeyExtractor()));
+    builder.connectVertices(newEdge);
+
+    // launch DAG
+    JobLauncher.launchDAG(builder.build());
+
+    // Retrieve result data from file.
+    try {
+      final Kryo kryo = new Kryo();
+      final List<T> result = new ArrayList<>();
+      Integer i = 0;
+      File file = new File(resultFile + i);
```

Comment:
Leave a comment to explain why `new File(resultFile + i)` is expected to exist at this point in the code.

Reply:
It's to be removed with #711. I'll add a note.